targets/targets.json

```diff
@@ -2801,6 +2801,7 @@
                 "value": "0xFFFFFFFF"
             }
         },
+        "OUTPUT_EXT": "bin",
```

Comment:
to switch support to hex file output, this needs to be changed to "OUTPUT_EXT": "hex".  even if the post build script takes the bin and converts it to hex, you still need to make this setting so that the online compiler will know to download the hex to the target and not the bin.  this setting does require changes to the post build script.